main/lsp/requests/completion.cc

```diff
@@ -483,6 +489,100 @@ unique_ptr<CompletionItem> getCompletionItemForLocalName(const core::GlobalState
     return item;
 }
 
+vector<core::NameRef> allSimilarFields(const core::GlobalState &gs, core::ClassOrModuleRef klass, string_view prefix) {
+    vector<core::NameRef> result;
+
+    // `ancestors` already includes klass, so we don't have to handle klass specially
+    // as we do in allSimilarConstantItems.
+    for (auto ancestor : ancestors(gs, klass)) {
+        for (auto [name, sym] : ancestor.data(gs)->members()) {
+            if (!sym.isFieldOrStaticField()) {
+                continue;
+            }
+
+            // TODO: this does prefix matching for instance/class variables, but our
+            // completion for locals matches anywhere in the name
+            if (hasPrefixedName(gs, name, prefix)) {
+                result.emplace_back(name);
+            }
```

Comment:
My sense (possibly informed too much by my own personal idiosyncrasies) is that people would want prefix matching on instance (resp. class) variables, but I'm happy to hear alternative points of view.